src/local_pathfinding/local_pathfinding/obstacles.py

```diff
@@ -101,7 +100,132 @@ def update_reference_point(self, reference: HelperLatLon):
 
         if isinstance(self, Boat):
             # regenerate collision zone with updated reference point
-            self.update_boat_collision_zone()
+            self._update_boat_collision_zone()
+
+        else:
+            # regenerate collision zone with updated reference point
+            self._update_land_collision_zone()  # type: ignore
+
+
+class Land(Obstacle):
+    """
+    Describes land and territorial waters that which Sailbot must avoid. During runtime,
+    the obstacle tracker will keep track of a single Land obstacle object and update its collision
+    zone with new geometry when required.
+
+    Attributes:
+        collision_zone (MultiPolygon): A collection of Polygons in (X,Y) that define regions of
+                                       land stored in the Land object.
+        next_waypoint (HelperLatLon): Lat/lon position of the next global waypoint.
```

Comment:
Doesn't the next global waypoint already exist as `reference` in the parent class?

Reply:
YES! good point, I'll remove that next_waypoint parameter and use reference in its place.